libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -77,7 +77,11 @@ public struct RestResponse {
     /// Decode the response as  a codable.
     /// - Parameter type: The type to use for decoding.
     public func asDecodable<T: Decodable>(type: T.Type) throws -> Decodable? {
+        let dateFormatter = DateFormatter()
```

Comment:
@woxtu We should account for the fact that date format could occur in a  different format through query or maybe org locale prefs?  
e.g.```SELECT Id, LastModifiedDate, FORMAT(LastModifiedDate) formattedDate FROM Account ``` .   In fact my preference would be to pass in the JSONDecoder() itself. We should perhaps add something like  ```asDecodable(type: T.Type,  decoder: JSONDecoder)``` and this should let developers control decoding requirements not just for date but for other types as well.

Reply:
In fact I think doing this is probably sufficient? 
```public func asDecodable<T: Decodable>(type: T.Type, decoder: JSONDecoder? = JSONDecoder()) throws -> T {```

Reply:
@trooper2013 Thank you for pointing that out. It seems better to pass a decoder, so I'll create a new PR later.